help/Readme.md

```diff
@@ -0,0 +1,114 @@
+
+#### Service principal
+Packer uses Service Principal to authorize in Azure infrastructure. To setup image-generation CI or use packer manually — SP with full read-write permissions for selected Azure subscription needed.
+Detailed instruction can be found [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+### Prepare environment and image deployment
+#### How to prepare Windows build agent
+You can use local machine as build agent, or deploy virtual machine in [Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/quick-create-cli).
+
+Download `packer` from https://www.packer.io/downloads, or install it via Chocolately.
+```
+Choco-Install -PackageName packer
+```
+
+Install Azure CLI - https://docs.microsoft.com/ru-ru/cli/azure/install-azure-cli-windows?view=azure-cli-latest&tabs=azure-cli.
+```
+Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; rm .\AzureCLI.msi
+```
+
+Download Virtual-Environments repository.
+```
+git clone https://github.com/actions/virtual-environments.git
+```
+
+Import [GenerateResourcesAndImage](https://github.com/actions/virtual-environments/blob/main/helpers/GenerateResourcesAndImage.ps1) script from `/helpers` folder, and run `GenerateResourcesAndImage` function in Powershell.
+
+```
+Import-Module C:\virtual-environments\helpers\GenerateResourcesAndImage.ps1
+
+GenerateResourcesAndImage -SubscriptionId {YourSubscriptionId} -ResourceGroupName "myTestResourceGroup" -ImageGenerationRepositoryRoot "C:\virtual-environments" -ImageType Ubuntu1604 -AzureLocation "East US"
+```
+Where:
+- `SubscriptionId` - The Azure subscription Id where resources will be created.
+- `ResourceGroupName` - The Azure resource group name where the Azure resources will be created.
+- `ImageGenerationRepositoryRoot` - The root path of the image generation repository source.
+- `ImageType` - The type of the image being generated. Valid options are: "Windows2016", "Windows2019", "Ubuntu1604", "Ubuntu1804", "Ubuntu2004".
+- `AzureLocation` - The location of the resources being created in Azure. For example "East US".
+
+Function will automatically create all required Azure resources and kick off a packer image generation for the selected image type.
+
+*Please, check synopsis of `GenerateResourcesAndImage` for details about non mandatory parameters.*
+
+#### Generated VM Deployment
+After successful image generation, you can deploy Virtual Machine based on generated VHD using [CreateAzureVMFromPackerTemplate](https://github.com/actions/virtual-environments/blob/main/helpers/CreateAzureVMFromPackerTemplate.ps1) script.
```

Comment:
#### Generated VM Deployment
After the successful image generation, Virtual Machine can be created from the generated VHD using [CreateAzureVMFromPackerTemplate](https://github.com/actions/virtual-environments/blob/main/helpers/CreateAzureVMFromPackerTemplate.ps1) script.

Reply:
Done. Thanks!